### PR TITLE
Add GeometryTypeConverter

### PR DESCRIPTION
**Pull request by rabbitism**
## What does the pull request do?

Add GeometryConverter to support Path.Data Binding to string directly, which is supported in WPF but missing in Avalonia.

## What is the current behavior?

Currently reports Binding Error.

## How was the solution implemented (if it's not obvious)?

Added GeometryConverter.
Notice that StreamGeometry data is not stored anywhere, so it's impossible to convert back from Geometry to string. 

## Checklist

- [x] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by workgroupengineering** [comment]
Add support at compile time like this

https://github.com/workgroupengineering/Avalonia/blob/5e52a5989755486c541127f58a45c3270db04add/src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AvaloniaXamlIlLanguageParseIntrinsics.cs#L114-L129

**Comment by rabbitism** [comment]
@workgroupengineering there must be something I missed, but Avalonia already supports string -> geometry at compile time. I cannot figure out how it works.

**Comment by maxkatz6** [comment]
Currently following code:
```xaml
<Path Data="M406.39,333.45l205.93,0" />
```
is not compile-time parsed. And relevant compiled code looks like:
```csharp
		Path path;
		Path path2 = (path = new Path());
		((ISupportInitialize)path2).BeginInit();
		P_1.Content = path2;
		path.Data = Geometry.Parse("M406.39,333.45l205.93,0");
		((ISupportInitialize)path).EndInit();
```
It is possible to add compile time parsing as @workgroupengineering said. It just won't be that easy, as whole geometry construction process needs to be replicated in IL code.

But I am not sure if that issue is relevant to this PR. With binding from string to geometry runtime parsing always will be involved. There is no way to predict what string input will be before hands.
So, this GeometryConverter makes sense anyway.

File: src/Avalonia.Base/Media/Geometry.cs [hunk 1]
```diff
@@ -199,4 +202,32 @@ public static Geometry Combine(Geometry geometry1, RectangleGeometry geometry2,
             return new CombinedGeometry(combineMode, geometry1, geometry2, transform);
         }
     }
+
+    public class GeometryConverter : TypeConverter
```
Comment (maxkatz6) [inline src/Avalonia.Base/Media/Geometry.cs:206]: For consistency, can it be renamed to `GeometryTypeConverter` ?

**Comment by maxkatz6** [comment]
Also, as @rabbitism mentioned, compile time conversion from string to Geometry kinda actually exists, just not exactly what @workgroupengineering meant.
@workgroupengineering was suggesting compile time parsing. When in Avalonia we have compile time injecting of Geometry.Parse method.
For instance:
```csharp
public static string A => "M406.39,333.45l205.93,0";
```
```xaml
<Path Data="{x:Static local:Program.A}" />
```
will produce:
```
		Path path;
		Path path2 = (path = new Path());
		((ISupportInitialize)path2).BeginInit();
		P_1.Content = path2;
		path.Data = Geometry.Parse(Program.A); // implicitly parsed
		((ISupportInitialize)path).EndInit();
```

File: src/Avalonia.Base/Media/Geometry.cs (hunk 1 above)
Comment (maxkatz6) [reply src/Avalonia.Base/Media/Geometry.cs:206]: Also, can you add a single test for this case? @rabbitism

Comment (maxkatz6) [reply src/Avalonia.Base/Media/Geometry.cs:206]: Other than that, I don't think compile time parsing needs to be done in this PR. Maybe in the future, if somebody wants to.

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028473-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**